packages/flutter/lib/src/cupertino/app.dart

```diff
@@ -104,6 +104,50 @@ class CupertinoApp extends StatefulWidget {
        assert(checkerboardOffscreenLayers != null),
        assert(showSemanticsDebugger != null),
        assert(debugShowCheckedModeBanner != null),
+       routeInformationParser = null,
+       routerDelegate = null,
+       backButtonDispatcher = null,
+       routeInformationProvider = null,
+       super(key: key);
+
+  /// Creates a [CupertinoApp] that uses the [Router] instead of a [Navigator].
+  const CupertinoApp.router({
+    Key key,
+    @required this.routeInformationParser,
+    @required this.routerDelegate,
+    this.backButtonDispatcher,
+    this.routeInformationProvider,
```

Comment:
I find it weird to wrap backButtonDispatcher and routeInformationProvider together because they are pretty much standalone objects.

I am ok with wrap both route name parser and router delegate together, but I couldn't figure out what will be the api looks like if we provide an opinionated one.